tests/test_jss.py

```diff
@@ -6,114 +6,175 @@
-        "sequential",
-        "revsequential",
-        "pairwise",
-        "revpairwise",
-    ]:
-        for b in ["democracy", "continent"]:
-            c = comparisons(
-                m,
-                by=b,
-                variables={"equal": [30, 90]},
-                hypothesis=h,
-            )
-            assert isinstance(c, pl.DataFrame)
-            if b == "democracy":
-                assert c.shape[0] == 1
-            else:
-                assert c.shape[0] > 1
-
-
-def test_transform():
-    c1 = avg_comparisons(m, comparison="lnor")
-    c2 = avg_comparisons(m, comparison="lnor", transform=np.exp)
+@pytest.mark.parametrize(
+    "h, label",
+    [
+        (
+            "reference",
+            {
+                "democracy": "Democracy - Autocracy",
+                "continent": ["Americas - Africa", "Asia - Africa", "Europe - Africa"],
+            },
+        ),
+        (
+            "revreference",
+            {
+                "democracy": "Autocracy - Democracy",
+                "continent": ["Africa - Americas", "Africa - Asia", "Africa - Europe"],
+            },
+        ),
+        (
+            "sequential",
+            {
+                "democracy": "Democracy - Autocracy",
+                "continent": ["Americas - Africa", "Asia - Americas", "Europe - Asia"],
+            },
+        ),
+        (
+            "revsequential",
+            {
+                "democracy": "Autocracy - Democracy",
+                "continent": ["Africa - Americas", "Americas - Asia", "Asia - Europe"],
+            },
+        ),
+        (
+            "pairwise",
+            {
+                "democracy": "Autocracy - Democracy",
+                "continent": [
+                    "Africa - Americas",
+                    "Africa - Asia",
+                    "Africa - Europe",
+                    "Americas - Asia",
+                    "Americas - Europe",
+                    "Asia - Europe",
+                ],
+            },
+        ),
+        (
+            "revpairwise",
+            {
+                "democracy": "Democracy - Autocracy",
+                "continent": [
+                    "Americas - Africa",
+                    "Asia - Africa",
+                    "Europe - Africa",
+                    "Asia - Americas",
+                    "Europe - Americas",
+                    "Europe - Asia",
+                ],
+            },
+        ),
+    ],
+)
+def test_hypothesis_shape_and_row_labels(h, label, impartiality_model):
+    for b in ["democracy", "continent"]:
+        c = comparisons(
+            impartiality_model,
+            by=b,
+            variables={"equal": [30, 90]},
+            hypothesis=h,
+        )
+        assert isinstance(c, pl.DataFrame)
+        if b == "democracy":
+            assert c.shape[0] == 1
+            assert c["term"][0] == label[b]
+        else:
+            assert c.shape[0] > 1
+            assert (c["term"] == label[b]).all()
```

Comment:
@vincentarelbundock do you agree on testing the labels here?

Reply:
yes, that's fine.